godot-macros/src/derive_godot_class.rs

```diff
@@ -321,3 +418,32 @@ fn make_exports_impl(class_name: &Ident, fields: &Fields) -> TokenStream {
         }
     }
 }
+
+/// Checks at compile time that a function with the given name exists on `Self`.
+#[must_use]
+fn existence_check(ident: &Ident) -> TokenStream {
+    quote! {
+        #[allow(path_statements)]
+        Self::#ident;
+    }
+}
+
+/// Tests that the proc-macro doesn't panic. This is helpful because `RUST_BACKTRACE=1` does not
+/// affect panics in macro invocations (e.g. in our integration tests), but it works fine in unit
+/// tests.
+#[test]
+fn does_not_panic() {
+    let decl = venial::parse_declaration(quote! {
+        #[class(base=Node)]
+        struct HasProperty {
+            #[base]
+            base: Base<Node>,
+            #[export]
+            int_val_default: i32,
+            #[export(get = "get_int_val", set = "set_int_val")]
+            int_val: i32,
+        }
+    })
+    .unwrap();
+    transform(decl).unwrap();
+}
```

Comment:
Not sure I understand the background of this test, despite the description. Isn't this already fully covered by integration tests, or what are we testing here? `transform()` is also not part of any public API.

Reply:
Ah, I see. That's a cool site by the way! 👍